compiler/passes/normalize.cpp

```diff
@@ -2605,7 +2632,23 @@ static void hack_resolve_types(ArgSymbol* arg) {
       // dtUnknown or dtAny, then replace the type expression with that type.
       // hilde sez: don't we lose information here?
       if (arg->typeExpr->body.length == 1) {
-        Type* type = arg->typeExpr->body.only()->typeInfo();
+        Expr* only = arg->typeExpr->body.only();
+        Type* type = only->typeInfo();
+
+        // The type constructor might be known, but we don't want to remove
+        // the call to it just yet. See 'normalizeCallToTypeConstructor' for
+        // more information on how a SymExpr could be the baseExpr at this
+        // stage.
```

Comment:
This feels work-around-y but it's not obvious to me how else it could be written. Probably worth thinking about if there's a "cleaner" way to do it for 5 minutes (or something). If we keep this strategy, it might be worth more clearly labeling it as a workaround. Or, why should this apply only to functions with FLAG_TYPE_CONSTRUCTOR?
Why does the call to the type constructor need to remain? Does this have to do with generic types? Could the comment or the code make the root issue clearer?

Reply:
Minor update: I tried reworking this section to check whether 'type' was generic, but ran into some cryptic issue with tuples. I'm just going to update the comment.